uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/absence-tab/components/annual-entitlement-change-log.component.js

```diff
@@ -0,0 +1,319 @@
+/* eslint-env amd */
+
+define([
+  'common/lodash',
+  'common/moment',
+  'leave-absences/absence-tab/modules/components',
+  'common/models/contract'
+], function (_, moment, components) {
+  components.component('annualEntitlementChangeLog', {
+    bindings: {
+      contactId: '<',
+      periodId: '<',
+      dismissModal: '&'
+    },
+    templateUrl: ['settings', function (settings) {
+      return settings.pathTpl + 'components/annual-entitlement-change-log.html';
+    }],
+    controllerAs: 'changeLog',
+    controller: AnnualEntitlementChangeLog
+  });
+
+  AnnualEntitlementChangeLog.$inject = ['AbsencePeriod', 'AbsenceType',
+    'Entitlement', 'OptionGroup', 'shared-settings'];
+
+  function AnnualEntitlementChangeLog (AbsencePeriod, AbsenceType, Entitlement,
+  OptionGroup, sharedSettings) {
+    var entitlementsChangeLog = [];
+    var vm = this;
+
+    vm.absencePeriod = null;
+    vm.absenceTypes = [];
+    vm.changeLogRows = [];
+    vm.loading = { component: true };
+
+    (function init () {
+      loadAbsencePeriod()
+      .then(loadCalculationUnits)
+      .then(loadAbsenceTypes)
```

Comment:
You may not need this, please see the comments above about Absence Type loadCalculationUnits()